src/worldmap.cc

```diff
@@ -3281,6 +3290,8 @@ static int wmWorldMapFunc(int a1)
 
     wmFadeIn();
 
+    inWorldMap = false;
```

Comment:
@cambragol we're already tracking the current game mode using `GameMode::getCurrentGameMode()`, so we can remove this